certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -210,8 +219,8 @@ def prepare(self):
             raise errors.PluginError('Unable to lock {0}'.format(self.conf('server-root')))
 
     # Entry point in main.py for installing cert
-    def deploy_cert(self, domain, cert_path, key_path,
-                    chain_path=None, fullchain_path=None):
+    def deploy_cert(self, domain: str, cert_path: str, key_path: str,
+                    chain_path: Optional[str] = None, fullchain_path: Optional[str] = None) -> None:
```

Comment:
I typed based on the original signature of the configurator class here, but I agree with you that the interface do not declare `chain_path` and `fullchain_path` as `Optional[str]` and maybe more importantly, they are supposed to be positional.

I avoided as much as possible to modify the actual behavior of the code to start with, because it would have made a lot harder the typing if the errors can come from the typing itself or mistake I introduce by modifying the actual behavior.

Now that it is settled, indeed we can start to talk about the discrepencies that the typing is revealing. Here it is one, the signature of the configurator do not represent the actual usages: `chain_path` and `fullchain_path` are always set. Moreover the implementation shows that it is not expected to be something else. I also checked the `install` phase, and the others, that come down to the `certbot._internal.client.install_certificate()` function that always set the `chain_path` and the `fullchain_path` to the underlying intaller plugin.

This and the fact `certbot-nginx` code is not part of the public API convince me to modify now the signatures and make them coherent with the interfaces.